File: MovieFavorite/app/src/main/java/com/moviefavorite/DetailActivity.java
package com.moviefavorite;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.bumptech.glide.Glide;

import java.util.Locale;

public class DetailActivity extends AppCompatActivity {
    DatabaseHelper databaseHelper;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail);
        Bundle b = getIntent().getExtras();
        ImageView imageview = findViewById(R.id.imageDetail);
        TextView titleview  = findViewById(R.id.titleDetail);
        TextView avgview  = findViewById(R.id.voteDetail);
        TextView descview  = findViewById(R.id.descDetail);
        Button save = findViewById(R.id.btnSave);
        databaseHelper = new DatabaseHelper(this);
        titleview .setText(b.getString("title"));
        avgview .setText(getResources().getText(R.string.va).toString()+" : "+b.getDouble("avg"));
        descview .setText(b.getString("desc"));
        if(b.getString("fav")!=null){
            save.setVisibility(View.GONE);
        }
        Glide.with(this)
                .load("https://image.tmdb.org/t/p/w500"+b.getString("img"))
                .into(imageview);
        save.setOnClickListener(view -> {
            if(databaseHelper.insertMovie(
                    b.getString("title"),
                    b.getString("img"),
                    b.getDouble("avg"),
                    b.getString("desc"))){
                Toast.makeText(this, getResources().getText(R.string.add_success).toString(),
                        Toast.LENGTH_LONG).show();
            }else{
                Toast.makeText(this, getResources().getText(R.string.error).toString(),
                        Toast.LENGTH_LONG).show();
            }
        });

    }
}

File: Android/Kotlin/StoryApp/app/src/main/java/com/dicoding/picodiploma/storyapp/DetailActivity.kt
package com.dicoding.picodiploma.storyapp

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.bumptech.glide.Glide
import com.dicoding.picodiploma.storyapp.databinding.ActivityDetailBinding

class DetailActivity : AppCompatActivity() {

    private lateinit var binding: ActivityDetailBinding
    companion object {
        const val DETAIL_ACTIVITY = "detail_activity"
    }
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityDetailBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val stories = intent.getParcelableExtra<StoryList>(DETAIL_ACTIVITY) as StoryList
        Glide.with(this)
            .load(stories.photo)
            .into(binding.imgDetail)
        binding.tvNameDetail.text = stories.name
        binding.tvDescriptionDetail.text = stories.description

    }

}